### PR TITLE
Copy DOB js function to payoluton.js

### PR DESCRIPTION
**Pull request by JEreth**
Copy function that extracts dob from select fields to hidden text field to payolution.js to make it independent from ratepay.js. If this is not done and one uses payolution without ratepay the dob will not be collected.

### CODE REVIEW

**Comment by JEreth** [comment]
also added the same for the function copyDebitPaymentSepaIban from debitpayment.js

**Comment by fjbender** [comment]
@p1fb Could you review this?

File: js/payone/core/payolution.js [hunk 1]
```diff
@@ -157,4 +157,23 @@ function switchInstallmentPlan(sKey, sCode, iInstallments)
...
     $(sCode + '_selected_installmentplan').value = iInstallments;
+}
+
+/**
+ *
+ * @param payment_code
+ */
+function payonePayolutionCustomerDobInput(payment_code)
+{
+    var daySelect = $(payment_code + '_additional_fields_customer_dob_day');
```
Comment (p1fb) [inline js/payone/core/payolution.js]: Perhaps the # for jQuery ID selector is missing. Can you double check this?

File: js/payone/core/payolution.js [hunk 2]
```diff
@@ -157,4 +157,23 @@ function switchInstallmentPlan(sKey, sCode, iInstallments)
...
+    var monthSelect = $(payment_code + '_additional_fields_customer_dob_month');
+    var yearSelect = $(payment_code + '_additional_fields_customer_dob_year');
+    var hiddenDobFull = $(payment_code + '_additional_fields_customer_dob_full');
+
+    if (daySelect == undefined || monthSelect == undefined || yearSelect == undefined
```
Comment (p1fb) [inline js/payone/core/payolution.js]: A non-existing jQuery selector won't result in an undefined value. The return value will be a jQuery object which length is zero. The condition always evaluates to false.

File: js/payone/core/payolution.js [hunk 3]
```diff
@@ -157,4 +157,23 @@ function switchInstallmentPlan(sKey, sCode, iInstallments)
...
+        || hiddenDobFull == undefined)  {
+        return;
+    }
+
+    hiddenDobFull.value = yearSelect.value + "-" + monthSelect.value + "-" + daySelect.value;
```
Comment (p1fb) [inline js/payone/core/payolution.js]: The jQuery object provides the .val() method to retrieve the value. Nevertheless .value property should work too. Maybe an additional check for valid values could make sense.

Comment (JEreth) [reply js/payone/core/payolution.js]: The function is perfectly fine. The issue is simply that the files ratepay.js or debitpayment.js will not be loaded when the belonging payment methods are not active. Hence, the function is undefined. (In the concrete scenario we only use payolution without ratepay and directdebit methods.)

**Comment by hreinberger** [comment]
works! thx @JEreth!

**Pull request merged**